### ISSUE TITLE
Extremely slow type checking of code involving pandas arithmetic

### ISSUE THREAD

**Issue by hamdanal** [python/mypy#21017]
This snippet takes more than 6s to check (with `--no-incremental`)

```py
import pandas as pd

df = pd.DataFrame({"a": [1, 2, 3], "b": [4, 5, 6], "c": [7, 8, 9]})
df["d"] = df["a"] + df["b"] + df["c"] + 1 + (df["a"] ** 2 + df["b"] ** 2 + df["c"] ** 2)
```

Note that pandas-stubs has a ton of overloads for binary operator methods ([example](https://github.com/pandas-dev/pandas-stubs/blob/65b2772ed5da9c3a67e4d608ffd537a02656c398/pandas-stubs/core/series.pyi#L1754-L1841)) that contribute to the problem.

Versions:
```
mypy          1.19.1
numpy         2.4.3
pandas        3.0.1
pandas-stubs  3.0.0.260204
```

**Comment by ilevkivskyi** [comment]
I don't think the overloads are  the problem. I just tried comparing time for type-checking a file with single `import pandas as pd` vs your code (even with these two lines copy-pasted 25 times), and I don't see any visible difference:
```
$ time mypy -c "import pandas as pd"
Success: no issues found in 1 source file

real	0m3.208s
user	0m3.023s
sys	0m0.164s
```
vs
```
time mypy example.py
Success: no issues found in 1 source file

real	0m3.239s
user	0m3.061s
sys	0m0.176s
```
(compiled version from a recent master). The problem is that pandas is a huge library, so it simply takes a lot to analyse it. Eager processing of 3rd party libraries is a known issue (search issue tracker for `torch`), but there is not much we can do ATM.
